Dialog/index.d.ts

```diff
@@ -0,0 +1,180 @@
+import MaterialComponent from '../MaterialComponent';
+import { VNode, PreactHTMLAttributes } from 'preact';
+import Button from '../Button';
+import { MDCFoundation, MDCComponent } from '../MaterialComponentsWeb';
+
+declare interface IDialogProps extends JSX.HTMLAttributes {
+  onAccept?: () => void;
+  onCancel?: () => void;
+}
+export default class Dialog extends MaterialComponent<IDialogProps, {}> {
+  static Header: typeof Header;
+  static Body: typeof Body;
+  static Footer: typeof Footer;
+  static FooterButton: typeof FooterButton;
+
+  MDComponent: MDCDialog;
+}
+
+declare class Header extends MaterialComponent<JSX.HTMLAttributes, {}> {}
+
+declare interface IBodyProps extends JSX.HTMLAttributes {
+  scrollable?: boolean;
+}
+declare class Body extends MaterialComponent<IBodyProps, {}> {}
+
+declare class Footer extends MaterialComponent<JSX.HTMLAttributes, {}> {}
+
+declare interface IFooterButtonProps extends HTMLAttributes {
+  accept?: boolean;
+  cancel?: boolean;
+}
+declare class FooterButton extends Button<IFooterButtonProps, {}> {}
+
+declare class MDCDialogFoundation extends MDCFoundation<MDCDialog> {
+  open(): void;
+  close(): void;
+  isOpen(): boolean;
+  accept(shouldNotify: boolean): void;
+  cancel(shouldNotify: boolean): void;
+}
+
+declare class MDCDialog extends MDCComponent<MDCDialogFoundation> {
+  open(): boolean;
+  show(): void;
+  close(): void;
+}
+
+interface HTMLAttributes extends PreactHTMLAttributes, JSX.DOMAttributes {
```

Comment:
Not a fan of having to maintain these...
I guess they don't change that often but still would be nice to find a better way. Not sure that there is though.

But at least should name it something like `HTMLAttributesWithoutAccept` or add a comment about why this is needed and how it differs from the real HTMLAttributes etc.

Reply:
That makes sense. I will add the comments and rename them.